pkg/build/controller/build/defaults/defaults.go

```diff
@@ -79,11 +79,14 @@ func setPodLogLevelFromBuild(pod *corev1.Pod, build *buildv1.Build) error {
 }
 
 func (b BuildDefaults) applyPodDefaults(pod *corev1.Pod, isCustomBuild bool) {
-	if len(b.Config.NodeSelector) != 0 && pod.Spec.NodeSelector == nil {
-		// only apply nodeselector defaults if the pod has no nodeselector labels
-		// already.
+	// TODO this is needs to be fixed properly
```

Comment:
@adambkaplan this indicates that something is likely to be broken in the config mapping, but we need something to land the rest of the bumps.